components/tabs/tab.js

```diff
@@ -0,0 +1,47 @@
+import { css, html, LitElement, unsafeCSS } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { getFocusPseudoClass } from '../../helpers/focus.js';
+import { RtlMixin } from '../../mixins/rtl/rtl-mixin.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+import { TabMixin } from './tab-mixin.js';
+
+class Tab extends TabMixin(SkeletonMixin(RtlMixin(LitElement))) {
+
+	static styles = css`
+		.d2l-tab-text {
+			margin: 0.5rem;
+			overflow: hidden;
+			padding: 0.1rem;
+			text-overflow: ellipsis;
+			white-space: nowrap;
+		}
+		:host(:first-child) .d2l-tab-text {
+			margin-inline-end: 0.6rem;
+			margin-inline-start: 0;
+		}
+		:host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-text {
+			border-radius: 0.3rem;
+			box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+			color: var(--d2l-color-celestine);
+		}
+	`;
+
+	renderContent() {
+		const overrideSkeletonText = this.skeleton && (!this.text || this.text.length === 0);
```

Comment:
The component is currently relying on the consumer providing text in a `slot`, but this line is expecting them to provide a reactive `text` property.